src/app/features/reporting/sql-report/sql-report.service.spec.ts

```diff
@@ -262,4 +263,38 @@ describe("SqlReportService", () => {
       },
     ]);
   });
+  it("should convert sql type report into CSV string", () => {
+    const mockSqlData: SqlReportRow[] = [
+      {
+        key: "Total students",
+        value: 6,
+        level: 0,
+      },
+      {
+        key: "Students gender",
+        value: 10,
+        level: 0,
+      },
+      {
+        key: "Male",
+        value: 6,
+        level: 1,
+      },
+      {
+        key: "Female",
+        value: 4,
+        level: 1,
+      },
+    ];
+
+    const result = service.getCsvforV2(mockSqlData);
+
+    const expectedCsv = `Name,,Value
```

Comment:
why do we expect an empty column between these?

Reply:
In the getCsv method, `deepestLevel + 1` adds an empty column before "Value", which may be to ensure the alignment of value in the CSV output or "Value" column consistently(that's why we expected an empty column in the test cases). Still, I’m not sure why it's needed. Tom might have some idea about it.

Reply:
ah, I think this was done to give people a visual kind of indentation also in the downloaded file :+1: 

```suggestion
    // extra empty column to allow for indentation of name at child levels
    const expectedCsv = `Name,,Value
```